sources2csr/sources_reader.py

```diff
@@ -51,13 +55,44 @@ def read_configuration(config_dir) -> SourcesConfig:
         return SourcesConfig(**json.load(sources_config_file))
 
 
+def read_ngs_files(input_dir: str) -> List[NGS]:
+    """ Reads NGS files inside input_dir in 3 different formats: `.txt`, `.maf.gz` and `.seg`.
+
+    :param input_dir: NGS files directory
+    :return: Sequence of NGS objects
+    """
+    ngs_data = []
+    txt_reader = NgsTxtReader(input_dir)
+    maf_reader = NgsMafReader(input_dir)
+    seg_reader = NgsSegReader(input_dir)
+
+    for filename in NgsReader.list_files(input_dir):
+        # logger.debug('Parsing {} data file from {}'.format(filename, directory))
+        if filename.endswith('maf.gz'):
+            ngs_data += maf_reader.read_data(filename)
+        if filename.endswith('seg'):
+            ngs_data += seg_reader.read_data(filename)
+        elif filename.endswith('_all_data_by_genes.txt') or filename.endswith('_all_thresholded.by_genes.txt'):
+            ngs_data += txt_reader.read_data(filename)
+    return ngs_data
+
+
 class SourcesReader:
 
     def __init__(self, input_dir, output_dir, config_dir):
         self.input_dir = input_dir
         self.output_dir = output_dir
         self.sources_config = read_configuration(config_dir)
 
+    def read_ngs_data(self) -> Set[NGS]:
+        for parent_dir, dirs, files in os.walk(self.input_dir):
```

Comment:
Can't we just have as convention that NGS data is in `self.input_dir' + '/NGS'`?